Source/pfile.cpp

```diff
@@ -153,7 +153,8 @@ static HANDLE pfile_open_save_archive(DWORD save_num)
 
 static void pfile_SFileCloseArchive(HANDLE hsArchive)
 {
-	SFileCloseArchive(hsArchive);
+	if (hsArchive)

```

Comment:
Why do we call this function if the archive isn't open? I think we should `assert(hsArchive != NULL)` here instead

Reply:
We need to be able to safely call it multiple times, or more carefully track the state.
Or I don't know how to do this atm, feel free to send a PR that does this nicer.